### 1. Lanzamiento y Manejo de Múltiples Excepciones No Relacionadas

En situaciones donde es necesario reportar varias excepciones que han ocurrido, se puede usar la clase `ExceptionGroup` para agruparlas y manejarlas juntas.

**Ejemplo:**

```python
def f():
    excs = [OSError('error 1'), SystemError('error 2')]
    raise ExceptionGroup('hubo problemas', excs)

try:
    f()
except Exception as e:
    print(f'capturada {type(e)}: {e}')
```

### 2. Clase ExceptionGroup

`ExceptionGroup` es una excepción que envuelve una lista de instancias de excepciones, permitiendo que se lancen juntas y se manejen como una sola excepción.

**Ejemplo:**

```python
def f():
    excs = [OSError('error 1'), SystemError('error 2')]
    raise ExceptionGroup('hubo problemas', excs)

try:
    f()
except ExceptionGroup as eg:
    for exc in eg.exceptions:
        print(f'Excepción: {exc}')
```

### 3. Uso de except\* para Manejar Excepciones Específicas

Usando `except*` en lugar de except, se pueden manejar selectivamente solo las excepciones en el grupo que coincidan con un cierto tipo.

**Ejemplo:**

```python
def f():
    raise ExceptionGroup(
        "grupo1",
        [
            OSError(1),
            SystemError(2),
            ExceptionGroup(
                "grupo2",
                [
                    OSError(3),
                    RecursionError(4)
                ]
            )
        ]
    )

try:
    f()
except* OSError as e:
    print("Hubo errores de OSError")
except* SystemError as e:
    print("Hubo errores de SystemError")
```
